JumpscaleLibs/clients/stellar/StellarClient.py

```diff
@@ -508,3 +508,136 @@ def _set_account_signers(self, address, public_key_signer, preauth_tx_hash, sign
                 address=address, pk_signer=public_key_signer, preauth_hash_signer=preauth_tx_hash
             )
         )
+
+    def modify_signing_requirements(self, public_keys_signers, signature_count, low_treshold=1, high_treshold=2):
+        """modify_signing_requirements sets to amount of signatures required for the creation of multisig account. It also adds
+        the public keys of the signer to this account
+        :param public_keys_signers: list of public keys of signers.
+        :type public_keys_signers: list
+        :param signature_count: amount of signatures requires to transfer funds.
+        :type signature_count: str
+        :param low_treshold: amount of signatures required for low security operations (transaction processing, allow trust, bump sequence)
+        :type low_treshold: str
+        :param high_treshold: amount of signatures required for high security operations (set options, account merge)
+        :type: str
+        """
+        if len(public_keys_signers) != signature_count - 1:
+            raise Exception(
+                "Number of public_keys must be 1 less than the signature count in order to set the signature count"
+            )
+
+        server = self._get_horizon_server()
+        account = server.load_account(self.address)
+        source_keypair = Keypair.from_secret(self.secret)
+
+        transaction_builder = TransactionBuilder(account)
+        # set the signing options
+        transaction_builder.append_set_options_op(
+            low_threshold=low_treshold, med_threshold=signature_count, high_threshold=high_treshold
+        )
+
+        # For every public key given, add it as a signer to this account
+        for public_key_signer in public_keys_signers:
+            transaction_builder.append_ed25519_public_key_signer(public_key_signer, 1)
+
+        transaction_builder.set_timeout(30)
+        tx = transaction_builder.build()
+        tx.sign(source_keypair)
+
+        try:
+            response = server.submit_transaction(tx)
+            self._log_info(response)
+            self._log_info(
+                "Set the signers of {address} to require {signature_count} signers".format(
+                    address=self.address, signature_count=signature_count
+                )
+            )
+        except BadRequestError:
+            self._log_info("Transaction need additional signatures in order to send")
+            return tx.to_xdr()
+
+    def create_multisig_transaction(self, destination_address, amount, asset="XLM", memo_text=None):
```

Comment:
I think the name is too generic, it's a transfer, not just a generic transaction, I also wonder if it is not possible to check in the regular transfer if multiple signers are required

Reply:
Okay I changed the `transfer` method to return a transaction in XDR in case there are other signatures required and return an exception if the the operation is underfunded